applications/conceptual-model-editor/src/app/diagram/visualization.tsx

```diff
@@ -83,12 +84,13 @@ export const Visualization = () => {
     const nodeTypes = useMemo(() => ({ classCustomNode: ClassCustomNode }), []);
     const edgeTypes = useMemo(() => ({ floating: SimpleFloatingEdge }), []);
 
-    // TODO: Remove the next few comments if no issues will be found within next few days/weeks. And also remove the onInit line in the return.
-    // I think that the wrapper using ReactflowInstance provider in page.tsx is enough, hopefully it doesn't break anything.
-    // The Old code:
-    //const [reactFlowInstance, setReactFlowInstance] = useState<ReactFlowInstance | null>(null);
     const reactFlowInstance = useReactFlow();
 
+    const { CanvasMenuOptions, openCanvasMenuOptions, isCanvasMenuOptionsOpen, canvasMenuXYPosition, setCanvasMenuXYPosition } = useCanvasMenuOptions();    
```

Comment:
We really need to start using controllers/services for components. We can discuss this together later.

Reply:
We can, but I am not sure if I understand, what exactly do you mean by the terms do you have some page with example from top of your head (don't want you to spend too much time on searching). 

I have seen controller in your preview of the new CME, but I am not sure if I understand exactly. Is it something that provides all the relevant data used for rendering, while again it is in different file from the actual component, i.e. the rendering part (as in the previous conversation)?